game/world/respawnobject.cpp

```diff
@@ -0,0 +1,189 @@
+#include "respawnobject.h"
+
+#include <Tempest/Log>
+
+#include "game/serialize.h"
+#include "world/world.h"
+#include "gothic.h"
+
+#include <vector>
+#include <map>
+
+using namespace Tempest;
+
+std::vector<RespawnObject> RespawnObject::respawnList = {};
+std::string RespawnObject::processedDayWorld = "";
+uint32_t RespawnObject::minDays = 2;  // Should be at least 1
+uint32_t RespawnObject::maxDays = 10; // Must be bigger than minDays
+
+RespawnObject::RespawnObject(Serialize &fin){
+    fin.read(inst,wp,respawnDay,world);
+}
+
+void RespawnObject::save(Serialize &fout) const{
+    fout.write(inst,wp,respawnDay,world);
+}
+
+void RespawnObject::registerObject(size_t inst, std::string wp, uint32_t guild){
```

Comment:
Path seeming passes string by value a lot with little reason to do so.
And also performs copy, instead of move in this function.